engines/base/src/test/java/org/alfresco/transform/base/MtlsTestUtils.java

```diff
@@ -0,0 +1,69 @@
+package org.alfresco.transform.base;
+
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustAllStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.web.client.RestTemplate;
+
+import javax.net.ssl.SSLContext;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+
+public class MtlsTestUtils {
+
+    public static boolean isMtlsEnabled()
+    {
+        return Boolean.parseBoolean(System.getProperty("mtls-enabled"));
+    }
+
+    public static CloseableHttpClient httpClientWithMtls() throws NoSuchAlgorithmException, KeyManagementException, UnrecoverableKeyException, KeyStoreException, IOException, CertificateException {
+
+        String keyStoreFile = System.getProperty("mtls-keystore-file");
```

Comment:
"mtls-keystore-file" Where do these names come from? I don't really see them elsewhere in the code or in a properties file provided for tests. I'd expect "client.ssl...." or "server.ssl...." properties to be used

Reply:
Those are for test use only - specified in the pom.xml file for Transform Service end-to-end test:
`<systemPropertyVariables>
      <mtls-enabled>true</mtls-enabled>
      <mtls-keystore-file>./enterprise/integration-test/keystore.p12</mtls-keystore-file>
      <mtls-keystore-type>PKCS12</mtls-keystore-type>
      <mtls-keystore-password>password</mtls-keystore-password>
      <mtls-truststore-file>./enterprise/integration-test/truststore.p12</mtls-truststore-file>
      <mtls-truststore-type>PKCS12</mtls-truststore-type>
      <mtls-truststore-password>password</mtls-truststore-password>
  </systemPropertyVariables>`